### ISSUE TITLE
Kernel panic on El Capitan using 0.99.12 and Wineskin

### ISSUE THREAD

**Issue by tictactoes** [uglyDwarf/linuxtrack#103]
I'm using the Wiiserver for head tracking in IL2 1946 in a wineskin wrapper (v2.6.1) on a Mac and up until now it's been working very well. I'm guessing it has something to do with the recent update to El Capitan. Everything starts up as normal although the tracking takes a while before it starts. After a few minutes of play I get a kernel panic (see log below). If I disable linuxtrack the game runs as normal.

Wed Oct 21 13:57:25 2015

```
 *** Panic Report ***
panic(cpu 3 caller 0xffffff80139d6a9a): Kernel trap at 0xffffff8013d906f5, type 13=general protection, registers:
CR0: 0x000000008001003b, CR2: 0xffffff80f8d3e004, CR3: 0x00000002320bb0ea, CR4: 0x00000000001627e0
RAX: 0x0000000000000000, RBX: 0xdeadbeefdeadbeef, RCX: 0x0000000000000001, RDX: 0x0000000000000000
RSP: 0xffffff911ef53d00, RBP: 0xffffff911ef53d40, RSI: 0x0000000000000000, RDI: 0xffffff802ed776b0
R8:  0xffffff80140ddb78, R9:  0xffffff80140ddb78, R10: 0xffffff80140ddb78, R11: 0x0000000000000000
R12: 0xffffff802ed776b0, R13: 0xffffff802ed77750, R14: 0xffffff802ed776a8, R15: 0xffffff80370f8318
RFL: 0x0000000000010282, RIP: 0xffffff8013d906f5, CS:  0x0000000000000008, SS:  0x0000000000000010
Fault CR2: 0xffffff80f8d3e004, Error code: 0x0000000000000000, Fault CPU: 0x3, PL: 0

Backtrace (CPU 3), Frame : Return Address
0xffffff81156e5df0 : 0xffffff80138e5357 
0xffffff81156e5e70 : 0xffffff80139d6a9a 
0xffffff81156e6050 : 0xffffff80139f4093 
0xffffff81156e6070 : 0xffffff8013d906f5 
0xffffff911ef53d40 : 0xffffff8013b3ec46 
0xffffff911ef53d60 : 0xffffff8013b2dd90 
0xffffff911ef53de0 : 0xffffff8013b22bf0 
0xffffff911ef53e60 : 0xffffff8013dca287 
0xffffff911ef53f00 : 0xffffff8013dca0be 
0xffffff911ef53f60 : 0xffffff8013e2ace1 
0xffffff911ef53fb0 : 0xffffff80139f4896 

BSD process name corresponding to current thread: ltr_server1

Mac OS version:
15A284

Kernel version:
Darwin Kernel Version 15.0.0: Wed Aug 26 16:57:32 PDT 2015; root:xnu-3247.1.106~1/RELEASE_X86_64
Kernel UUID: 37BC582F-8BF4-3F65-AFBB-ECF792060C68
Kernel slide:     0x0000000013600000
Kernel text base: 0xffffff8013800000
__HIB  text base: 0xffffff8013700000
System model name: MacBookAir6,1 (Mac-35C1E88140C3E6CF)

System uptime in nanoseconds: 1736468215409
last loaded kext at 997062867601: com.apple.driver.AppleXsanScheme	3 (addr 0xffffff7f96389000, size 32768)
last unloaded kext at 1113203201082: com.apple.driver.AppleXsanScheme	3 (addr 0xffffff7f96389000, size 32768)
loaded kexts:
com.apple.filesystems.afpfs	11.0
com.apple.nke.asp-tcp	8.0.0
com.apple.driver.AudioAUUC	1.70
com.apple.driver.AGPM	110.20.19
com.apple.driver.ApplePlatformEnabler	2.5.1d0
com.apple.driver.X86PlatformShim	1.0.0
com.apple.filesystems.autofs	3.0
com.apple.driver.AppleHDA	272.50
com.apple.driver.AppleUpstreamUserClient	3.6.1
com.apple.driver.AppleOSXWatchdog	1
com.apple.driver.pmtelemetry	1
com.apple.iokit.IOUserEthernet	1.0.1
com.apple.iokit.IOBluetoothSerialManager	4.4.0f4
com.apple.driver.AppleLPC	3.1
com.apple.driver.AppleIntelHD5000Graphics	10.0.8
com.apple.iokit.BroadcomBluetoothHostControllerUSBTransport	4.4.0f4
com.apple.Dont_Steal_Mac_OS_X	7.0.0
com.apple.driver.AppleHV	1
com.apple.driver.AppleThunderboltIP	3.0.8
com.apple.driver.AppleIntelFramebufferAzul	10.0.8
com.apple.driver.AppleSMCLMU	208
com.apple.driver.AppleIntelSlowAdaptiveClocking	4.0.0
com.apple.driver.AppleCameraInterface	5.41.0
com.apple.driver.AppleBacklight	170.8.1
com.apple.driver.AppleMCCSControl	1.2.13
com.apple.driver.AppleUSBDisplays	378
com.apple.iokit.IOAHCIBlockStorage	2.8.0
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless	1.0.0d1
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib	1.0.0
com.apple.BootCache	37
com.apple.driver.AppleTopCaseHIDEventDriver	81.2
com.apple.driver.AppleAHCIPort	3.1.5
com.apple.driver.AirPort.Brcm4360	1000.13.1a3
com.apple.driver.AppleSmartBatteryManager	161.0.0
com.apple.driver.AppleRTC	2.0
com.apple.driver.AppleACPIButtons	4.0
com.apple.driver.AppleHPET	1.8
com.apple.driver.AppleSMBIOS	2.1
com.apple.driver.AppleACPIEC	4.0
com.apple.driver.AppleAPIC	1.7
com.apple.nke.applicationfirewall	163
com.apple.security.quarantine	3
com.apple.security.TMSafetyNet	8
com.apple.security.SecureRemotePassword	1.0
com.apple.kext.triggers	1.0
com.apple.driver.DspFuncLib	272.50
com.apple.kext.OSvKernDSPLib	525
com.apple.iokit.IOSurface	108.0.1
com.apple.iokit.IOBluetoothHostControllerUSBTransport	4.4.0f4
com.apple.iokit.IOBluetoothFamily	4.4.0f4
com.apple.driver.CoreCaptureResponder	1
com.apple.driver.AppleHDAController	272.50
com.apple.iokit.IOHDAFamily	272.50
com.apple.iokit.IOAudioFamily	203.8
com.apple.vecLib.kext	1.2.0
com.apple.AppleGraphicsDeviceControl	3.11.31
com.apple.iokit.IOAcceleratorFamily2	203.7.1
com.apple.driver.X86PlatformPlugin	1.0.0
com.apple.driver.IOPlatformPluginFamily	6.0.0d7
com.apple.iokit.IOSerialFamily	11
com.apple.iokit.IOSlowAdaptiveClockingFamily	1.0.0
com.apple.driver.AppleBacklightExpert	1.1.0
com.apple.iokit.IONDRVSupport	2.4.1
com.apple.driver.AppleSMC	3.1.9
com.apple.driver.AppleSMBusController	1.0.14d1
com.apple.iokit.IOGraphicsFamily	2.4.1
com.apple.iokit.IOSCSIArchitectureModelFamily	3.7.7
com.apple.driver.AppleUSBHIDMouse	185.3
com.apple.driver.AppleHIDMouse	185.3
com.apple.driver.AppleUSBComposite	900.4.1
com.apple.iokit.IOUSBHIDDriver	900.4.1
com.apple.iokit.IOUSBHIDDriverPM	900.4.1
com.apple.driver.CoreStorage	517
com.apple.driver.usb.IOUSBHostHIDDevice	1.0.1
com.apple.driver.usb.AppleUSBHostCompositeDevice	1.0.1
com.apple.driver.usb.AppleUSBHub	1.0.1
com.apple.driver.AppleMultitouchDriver	304.9
com.apple.driver.AppleHIDTransport	5
com.apple.driver.AppleHIDKeyboard	181
com.apple.driver.AppleHSSPIHIDDriver	41.6
com.apple.iokit.IOAHCIFamily	2.8.0
com.apple.driver.AppleThunderboltDPInAdapter	4.1.2
com.apple.driver.AppleThunderboltDPAdapterFamily	4.1.2
com.apple.driver.AppleThunderboltPCIDownAdapter	2.0.2
com.apple.driver.AppleThunderboltNHI	4.0.1
com.apple.iokit.IOThunderboltFamily	5.0.6
com.apple.iokit.IO80211Family	1100.23
com.apple.driver.mDNSOffloadUserClient	1.0.1b8
com.apple.iokit.IONetworkingFamily	3.2
com.apple.driver.corecapture	1.0.4
com.apple.driver.AppleHSSPISupport	41.6
com.apple.driver.AppleIntelLpssSpiController	2.0.57
com.apple.driver.AppleIntelLpssGspi	2.0.57
com.apple.driver.usb.AppleUSBXHCIPCI	1.0.1
com.apple.driver.usb.AppleUSBXHCI	1.0.1
com.apple.driver.AppleIntelLpssDmac	2.0.57
com.apple.iokit.IOUSBFamily	900.4.1
com.apple.iokit.IOUSBHostFamily	1.0.1
com.apple.driver.AppleUSBHostMergeProperties	1.0.1
com.apple.driver.AppleEFINVRAM	2.0
com.apple.driver.AppleEFIRuntime	2.0
com.apple.iokit.IOHIDFamily	2.0.0
com.apple.iokit.IOSMBusFamily	1.1
com.apple.security.sandbox	300.0
com.apple.kext.AppleMatch	1.0.0d1
com.apple.driver.AppleKeyStore	2
com.apple.driver.AppleMobileFileIntegrity	1.0.5
com.apple.driver.AppleCredentialManager	1.0
com.apple.driver.DiskImages	414
com.apple.iokit.IOStorageFamily	2.1
com.apple.iokit.IOReportFamily	31
com.apple.driver.AppleFDEKeyStore	28.30
com.apple.driver.AppleACPIPlatform	4.0
com.apple.iokit.IOPCIFamily	2.9
com.apple.iokit.IOACPIFamily	1.4
com.apple.kec.Libm	1
com.apple.kec.pthread	1
com.apple.kec.corecrypto	1.0
Model: MacBookAir6,1, BootROM MBA61.0099.B19, 2 processors, Intel Core i5, 1.3 GHz, 8 GB, SMC 2.12f143
Graphics: Intel HD Graphics 5000, Intel HD Graphics 5000, Built-In
Memory Module: BANK 0/DIMM0, 4 GB, DDR3, 1600 MHz, 0x80AD, 0x483943434E4E4E384A544D4C41522D4E544D
Memory Module: BANK 1/DIMM0, 4 GB, DDR3, 1600 MHz, 0x80AD, 0x483943434E4E4E384A544D4C41522D4E544D
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x117), Broadcom BCM43xx 1.0 (7.21.94.25.1a3)
Bluetooth: Version 4.4.0f4 16320, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD TS0256F, 251 GB
USB Device: USB 3.0 Bus
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: Hub
USB Device: Keyboard Hub
USB Device: Apple Optical USB Mouse
USB Device: Apple Keyboard
USB Device: Apple Cinema HD Display
Thunderbolt Bus: MacBook Air, Apple Inc., 23.6
```

**Comment by uglyDwarf** [comment]
Hello,
if you take a look at the Console (Applications/Utilities), do you see there a crash report for the ltr_server1 application? The thing is, that this report doesn't seem to contain an information during which call did the panic happened; unfortunately the dump doesn't contain info on which part of the kernel did the crash happened...

One more thing that might be interesting - if you try running ltr_pipe (it is in the .dmg), does that happen too?

Kind regards,

Michal

**Comment by tictactoes** [comment]
Hi,

Console I found this:

```
2015-10-21 11:25:29,057 ReportCrash[1029]: com.apple.message.domain: com.apple.crashreporter.writereport.crash
com.apple.message.signature: ltr_server1
com.apple.message.signature2: UNBUNDLED ||| ??? (???)
com.apple.message.signature3: 35E93650D1186911B652626D0ADD819A
com.apple.message.result: NO
com.apple.message.summarize: YES
SenderMachUUID: 2A505BF0-CE7C-3DB7-8755-86802FD69E3F

```

Does that help? I'm happy to run ltr_pipe too, but I'm afraid I don't know how.

**Comment by tictactoes** [comment]
Ok, I managed to figure out how to run ltd_pipe (doh;). The game launches as usual, tracking works right away, but after a minute or two I get a kernel panic again. I also found this reference in the system log, though I have no idea if it's related or not:

    Oct 23 13:36:30 Gustafs-MacBook-Air kernel[0]: process ltr_server1[10454] caught causing excessive wakeups. EXC_RESOURCE supressed due to audio playback

**Comment by jasso12** [comment]
I get kernel panics as well when playing X-Plane 10 with linuxtrack and TrackIR 5. Sometimes it happens almost immediately after I start flying and sometimes after a while, like 15-30 minutes or so. When I disable linuxtrack there are no crashes. Updating El Capitan to 10.11.1 didn't make any difference.

```
*** Panic Report ***
panic(cpu 3 caller 0xffffff80201d6bba): Kernel trap at 0xffffff80205909e5, type 13=general protection, registers:
CR0: 0x000000008001003b, CR2: 0x0000000103543010, CR3: 0x00000003c5466144, CR4: 0x00000000001627e0
RAX: 0x0000000000000000, RBX: 0xdeadbeefdeadbeef, RCX: 0x0000000000000001, RDX: 0x0000000000000000
RSP: 0xffffff92eababd00, RBP: 0xffffff92eababd40, RSI: 0x0000000000000000, RDI: 0xffffff8078276710
R8:  0xffffff80208e0b78, R9:  0xffffff80208e0dd0, R10: 0xffffff80208e0b78, R11: 0x0000000000000000
R12: 0xffffff8078276710, R13: 0xffffff80782767b0, R14: 0xffffff8078276708, R15: 0xffffff8064316d48
RFL: 0x0000000000010282, RIP: 0xffffff80205909e5, CS:  0x0000000000000008, SS:  0x0000000000000010
Fault CR2: 0x0000000103543010, Error code: 0x0000000000000000, Fault CPU: 0x3, PL: 0

Backtrace (CPU 3), Frame : Return Address
0xffffff82dd63ddf0 : 0xffffff80200e5307 
0xffffff82dd63de70 : 0xffffff80201d6bba 
0xffffff82dd63e050 : 0xffffff80201f4313 
0xffffff82dd63e070 : 0xffffff80205909e5 
0xffffff92eababd40 : 0xffffff802033ef86 
0xffffff92eababd60 : 0xffffff802032e0d0 
0xffffff92eababde0 : 0xffffff8020322f30 
0xffffff92eababe60 : 0xffffff80205ca577 
0xffffff92eababf00 : 0xffffff80205ca3ae 
0xffffff92eababf60 : 0xffffff802062afd1 
0xffffff92eababfb0 : 0xffffff80201f4b16 

BSD process name corresponding to current thread: ltr_server1

Mac OS version:
15B42

Kernel version:
Darwin Kernel Version 15.0.0: Sat Sep 19 15:53:46 PDT 2015; root:xnu-3247.10.11~1/RELEASE_X86_64
Kernel UUID: AB5FC1B4-12E7-311E-8E6F-9023985D8C1D
Kernel slide:     0x000000001fe00000
Kernel text base: 0xffffff8020000000
__HIB  text base: 0xffffff801ff00000
System model name: iMac14,2 (Mac-27ADBB7B4CEE8E61)

System uptime in nanoseconds: 24733231475242
last loaded kext at 24678115375864: com.apple.driver.AppleXsanScheme	3 (addr 0xffffff7fa324e000, size 32768)
last unloaded kext at 22629820486951: com.logitech.driver.LogiGamingMouseFilter	1 (addr 0xffffff7fa324e000, size 16384)
loaded kexts:
com.razer.common.razerhid	13.18
com.cyberic.SmoothMouse	9
com.driver.LogJoystick	2.0
at.obdev.nke.LittleSnitch	4352
com.3dconnexion.driver	10.2.4
com.apple.filesystems.afpfs	11.0
com.apple.nke.asp-tcp	8.0.0
com.apple.driver.AppleBluetoothMultitouch	90.3
com.apple.filesystems.smbfs	3.0.0
com.apple.driver.AudioAUUC	1.70
com.apple.driver.AppleHWSensor	1.9.5d0
com.apple.filesystems.ntfs	3.13
com.apple.filesystems.autofs	3.0
com.apple.driver.AGPM	110.20.21
com.apple.driver.ApplePlatformEnabler	2.5.1d0
com.apple.driver.X86PlatformShim	1.0.0
com.apple.driver.AppleMikeyHIDDriver	124
com.apple.driver.AppleOSXWatchdog	1
com.apple.driver.AppleMikeyDriver	272.50.31
com.apple.driver.AppleHDA	272.50.31
com.apple.iokit.BroadcomBluetoothHostControllerUSBTransport	4.4.2f1
com.apple.driver.pmtelemetry	1
com.apple.iokit.IOUserEthernet	1.0.1
com.apple.iokit.IOBluetoothSerialManager	4.4.2f1
com.apple.driver.AppleUpstreamUserClient	3.6.1
com.apple.driver.AppleThunderboltIP	3.0.8
com.apple.driver.AppleBacklight	170.8.1
com.apple.GeForce	10.0.6
com.apple.Dont_Steal_Mac_OS_X	7.0.0
com.apple.driver.AppleHV	1
com.apple.driver.AppleIntelHD5000Graphics	10.1.0
com.apple.driver.AppleLPC	3.1
com.apple.driver.AppleIntelSlowAdaptiveClocking	4.0.0
com.apple.driver.AppleMCCSControl	1.2.13
com.apple.driver.AppleSMCLMU	208
com.apple.driver.AppleIntelFramebufferAzul	10.1.0
com.apple.driver.initioFWBridge	3.7.1
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless	1.0.0d1
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib	1.0.0
com.apple.BootCache	37
com.apple.driver.AppleFWOHCI	5.5.2
com.apple.iokit.IOAHCIBlockStorage	2.8.0
com.apple.driver.AppleSDXC	1.7.0
com.apple.iokit.AppleBCM5701Ethernet	10.1.11
com.apple.driver.AirPort.Brcm4360	1010.14.1a4
com.apple.driver.AppleAHCIPort	3.1.5
com.apple.driver.AppleHPET	1.8
com.apple.driver.AppleACPIButtons	4.0
com.apple.driver.AppleRTC	2.0
com.apple.driver.AppleSMBIOS	2.1
com.apple.driver.AppleACPIEC	4.0
com.apple.driver.AppleAPIC	1.7
com.apple.nke.applicationfirewall	163
com.apple.security.quarantine	3
com.apple.security.TMSafetyNet	8
com.apple.driver.AppleXsanScheme	3
com.apple.security.SecureRemotePassword	1.0
com.apple.driver.AppleMultitouchDriver	304.9
com.apple.driver.AppleHIDTransport	5
com.apple.driver.AppleBluetoothHIDKeyboard	181
com.apple.driver.IOBluetoothHIDDriver	4.4.2f1
com.apple.driver.AppleHIDKeyboard	181
com.apple.kext.triggers	1.0
com.apple.driver.DspFuncLib	272.50.31
com.apple.kext.OSvKernDSPLib	525
com.apple.iokit.IOBluetoothHostControllerUSBTransport	4.4.2f1
com.apple.iokit.IOSerialFamily	11
com.apple.iokit.IOBluetoothFamily	4.4.2f1
com.apple.nvidia.driver.NVDAGK100Hal	10.0.6
com.apple.iokit.IOFireWireIP	2.2.6
com.apple.driver.AppleBacklightExpert	1.1.0
com.apple.nvidia.driver.NVDAResman	10.0.6
com.apple.driver.CoreCaptureResponder	1
com.apple.iokit.IOSurface	108.0.1
com.apple.driver.AppleHDAController	272.50.31
com.apple.iokit.IOHDAFamily	272.50.31
com.apple.iokit.IOAudioFamily	204.1
com.apple.vecLib.kext	1.2.0
com.apple.driver.AppleSMBusPCI	1.0.14d1
com.apple.driver.X86PlatformPlugin	1.0.0
com.apple.driver.AppleThunderboltEDMSink	4.1.1
com.apple.driver.IOPlatformPluginFamily	6.0.0d7
com.apple.iokit.IOSlowAdaptiveClockingFamily	1.0.0
com.apple.driver.AppleSMBusController	1.0.14d1
com.apple.driver.AppleSMC	3.1.9
com.apple.AppleGraphicsDeviceControl	3.11.33b1
com.apple.iokit.IOAcceleratorFamily2	203.10
com.apple.iokit.IONDRVSupport	2.4.1
com.apple.iokit.IOGraphicsFamily	2.4.1
com.apple.driver.usb.IOUSBHostHIDDevice	1.0.1
com.apple.driver.usb.AppleUSBHub	1.0.1
com.apple.driver.usb.AppleUSBHostCompositeDevice	1.0.1
com.apple.iokit.IOSCSIBlockCommandsDevice	3.7.7
com.apple.iokit.IOFireWireSerialBusProtocolTransport	2.5.1
com.apple.iokit.IOSCSIArchitectureModelFamily	3.7.7
com.apple.iokit.IOFireWireSBP2	4.2.7
com.apple.driver.CoreStorage	517
com.apple.driver.AppleThunderboltDPInAdapter	4.1.2
com.apple.driver.AppleThunderboltDPOutAdapter	4.1.2
com.apple.driver.AppleThunderboltDPAdapterFamily	4.1.2
com.apple.driver.AppleThunderboltPCIUpAdapter	2.0.2
com.apple.driver.AppleThunderboltPCIDownAdapter	2.0.2
com.apple.iokit.IOFireWireFamily	4.5.8
com.apple.driver.AppleThunderboltNHI	4.0.4
com.apple.iokit.IOThunderboltFamily	5.0.6
com.apple.iokit.IOEthernetAVBController	1.0.3b3
com.apple.iokit.IO80211Family	1101.24
com.apple.driver.mDNSOffloadUserClient	1.0.1b8
com.apple.iokit.IONetworkingFamily	3.2
com.apple.driver.corecapture	1.0.4
com.apple.iokit.IOAHCIFamily	2.8.0
com.apple.driver.usb.AppleUSBXHCIPCI	1.0.1
com.apple.driver.usb.AppleUSBXHCI	1.0.1
com.apple.driver.AppleEFINVRAM	2.0
com.apple.driver.AppleEFIRuntime	2.0
com.apple.iokit.IOSMBusFamily	1.1
com.apple.security.sandbox	300.0
com.apple.kext.AppleMatch	1.0.0d1
com.apple.driver.AppleKeyStore	2
com.apple.driver.AppleMobileFileIntegrity	1.0.5
com.apple.driver.AppleCredentialManager	1.0
com.apple.driver.DiskImages	415
com.apple.iokit.IOStorageFamily	2.1
com.apple.iokit.IOUSBHIDDriver	900.4.1
com.apple.iokit.IOUSBFamily	900.4.1
com.apple.iokit.IOUSBHostFamily	1.0.1
com.apple.driver.AppleUSBHostMergeProperties	1.0.1
com.apple.iokit.IOHIDFamily	2.0.0
com.apple.iokit.IOReportFamily	31
com.apple.driver.AppleFDEKeyStore	28.30
com.apple.driver.AppleACPIPlatform	4.0
com.apple.iokit.IOPCIFamily	2.9
com.apple.iokit.IOACPIFamily	1.4
com.apple.kec.Libm	1
com.apple.kec.pthread	1
com.apple.kec.corecrypto	1.0
Model: iMac14,2, BootROM IM142.0118.B12, 4 processors, Intel Core i5, 3,4 GHz, 24 GB, SMC 2.15f7
Graphics: NVIDIA GeForce GTX 780M, NVIDIA GeForce GTX 780M, PCIe, 4096 MB
Memory Module: BANK 0/DIMM0, 4 GB, DDR3, 1600 MHz, 0x02FE, 0x45424A3430554738454655352D474E4C2D46
Memory Module: BANK 1/DIMM0, 4 GB, DDR3, 1600 MHz, 0x02FE, 0x45424A3430554738454655352D474E4C2D46
Memory Module: BANK 0/DIMM1, 8 GB, DDR3, 1600 MHz, 0x0198, 0x393930353432382D3135352E413030472020
Memory Module: BANK 1/DIMM1, 8 GB, DDR3, 1600 MHz, 0x0198, 0x393930353432382D3135352E413030472020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x111), Broadcom BCM43xx 1.0 (7.21.94.29.1a4)
Bluetooth: Version 4.4.2f1 16391, 3 services, 27 devices, 1 incoming serial ports
Network Service: Ethernet, Ethernet, en0
PCI Card: pci11c1,5901, IEEE 1394 Open HCI, Thunderbolt@195,0,0
Serial ATA Device: APPLE SSD SD0128F, 121,33 GB
Serial ATA Device: APPLE HDD ST3000DM001, 3 TB
USB Device: USB 3.0 Bus
USB Device: Hub
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: FaceTime HD Camera (Built-in)
USB Device: EyeTV Hybrid
USB Device: Hub
USB Device: Joystick - HOTAS Warthog
USB Device: Razer Naga
USB Device: Throttle - HOTAS Warthog
USB Device: Composite Device
USB Device: Wacom Wireless Receiver
USB Device: MFG Crosswind V2
FireWire Device: INIC-1615P, Initio, Up to 800 Mb/sec
Thunderbolt Bus: iMac, Apple Inc., 23.10
Thunderbolt Device: Thunderbolt to FireWire Adapter, Apple Inc., 3, 5.6
```

**Comment by uglyDwarf** [comment]
Hello guys,
this looks like a MacOS X kernel bug... I thought it might be a faulty RAM, but this seems ver improbable now.

Stupid thing is, that for now I have no way to try and reproduce it (but I'll try to do something about it - I just can't promise when it will be done)...

I just wander - could you try to contact apple by following https://support.apple.com/en-us/HT201753 ?
Maybe they could come up with something...

I'm sorry I can't be of more help right now, but I'll try to rectify this ASAP...

Kind regards,

Michal

**Comment by jasso12** [comment]
Hello,
I disabled System Integrity Protection to see if it could be the cause, because it is a new feature in El Capitan that is supposed to protect system files. Unfortunately there is still panics with SIP disabled.

I will see if there is some place to report this to Apple using their forums (it is the only channel available it seems). I appreciate your efforts developing linuxtrack and I understand problems like these can be difficult to figure out.

**Comment by uglyDwarf** [comment]
Hello,
so now I'm able to reproduce the problem...

Now I wander if I can find what is going on...

Kind regards,

Michal

**Comment by jasso12** [comment]
Good to hear you can reproduce it too.

I might be off base here but I was thinking it might be some OS X background process that interferes somehow, because usually the crash happens after a while and not immediately. Last time there was a crash I looked at console log and time machine had started backup process few seconds before the panic. I disabled time machine and it happened again, but this time there was nothing in the console log prior the panic..

**Comment by uglyDwarf** [comment]
I don't know yet what might cause that kernel panic - the code is working both in Linux and OS X since Snow Leopard... Direct involvement of another process is somewhat unlikely due to the security reasons.

I guess I either do something that is not fully specified (like unlocking a mutex that is unlocked already) or there is just some bug in a branch of code that is not used very often...

I'm not sure if I'll be able to directly debug the problem as it seems to require having two Macs and also building a kernel with full debug info. I'll see how far will I be able to get (If I had only the annotated backtrace, it might help a lot).

Kind regards,

Michal

**Comment by tictactoes** [comment]
I would be happy to help if there is anything you want me to test or do. The problem is that I really have no idea what I'm doing so I would need very explicit instructions.

**Comment by uglyDwarf** [comment]
It is OK, it is my first kernel panic on Mac OS X, so I have to find a way to obtain a backtrace too...
Good news is, that I can reproduce it...

Will report back as soon as I have something new...

Kind regards,

Michal

**Comment by uglyDwarf** [comment]
OK, I have some news, although they aren't very good...

I tried to remove a daemon call from the ltr_server1 (long deprecated, so I gave it a shot) and that didn't cure the issue...

Tried looking for the Kernel debug kit for the current kernel, but the newest one available is for 10.11, buld 15A284, while what I have is 10.11.1 (15B42), so no joy here...

Then I thought OK, I'm going to build my own kernel with debug info, but the xnu source for the version I have is not available...

So for the moment I'm stuck until either debug kit or the source appears...

In the meantime, could either one of you try to send the report to Apple? The thing is that I have only Customac/hackintosh since my mini died and I'm not sure if they'd acept such a report...

I'll try to ask around on where I could get the symbols or kernel source and see if I can get somewhere.

Kind regards,

Michal

**Comment by uglyDwarf** [comment]
Diging a bit deeper, I finaly know where the problem happens (in the kernel)...
The annotated backtrace looks like this:

0xffffff809b1fddf0 : 0xffffff80038e5307      ffffff80002e5220 T _panic
0xffffff809b1fde70 : 0xffffff80039d6bba     ffffff80003d62a0 T _kernel_trap
0xffffff809b1fe050 : 0xffffff80039f4313      ffffff80003f4230 T _return_from_trap
0xffffff809b1fe070 : 0xffffff8003d909e5     ffffff80007909b0 T _knote
0xffffff80102a3d40 : 0xffffff8003b3ef86     ffffff800053ef30 T _fifo_write
0xffffff80102a3d60 : 0xffffff8003b2e0d0    ffffff800052e060 T _VNOP_WRITE
0xffffff80102a3de0 : 0xffffff8003b22f30     ffffff8000522440 T _vn_pathconf
0xffffff80102a3e60 : 0xffffff8003dca577    ??? some local unexported function
0xffffff80102a3f00 : 0xffffff8003dca3ae     ffffff80007ca2d0 T _write_nocancel
0xffffff80102a3f60 : 0xffffff8003e2afd1     ffffff800082ad80 T _unix_syscall64
0xffffff80102a3fb0 : 0xffffff80039f4b16     ffffff80003f4b00 T _hndl_unix_scall64

So the problem happens during writing to a fifo (so my guess is that it is in the master sending the pose data to the slave), in the knote function... Funny part is, that yesterday it took less than 10 seconds to panic, today it went for several minutes before crashing so that looks like some sort of race condition...

But what exactly is happening in there is anybody's guess (hard to tell without a peek to the sources, preferably comparing to the previous versions).

Kind regards,

Michal

PS. Just for the future reference - nm provides symbols, that can be roughly matched to the backtrace addresses, and otool -tV provides disassembly to confirm... And kernel is at /System/Library/Kernels ...

**Comment by uglyDwarf** [comment]
Good news - I have a plan to side-step that issue; I'm going to try to use sockets instead of the fifos...
If all goes well, hopefully during the weekend I should have something working.

Kind regards,

Michal

**Comment by tictactoes** [comment]
Great news! Your work is very much appreciated.

**Comment by jasso12** [comment]
Amazing news. I'm sorry I can't be much help fixing the problem as I'm not a real programmer and barely understand anything about this kernel stuff.

**Comment by uglyDwarf** [comment]
Short report on status: working on the change I found out that the consequences reach much deeper than I initially thought, which causes some slip in the schedule... Anyway, I'm progressing steadily and I'll keep you posted as soon as I have some news...

Kind regards,

Michal

**Comment by jasso12** [comment]
Thank you for the update. When you have the changes ready for testing let me know and I will hammer it in X-Plane.

**Comment by jasso12** [comment]
Maybe this has something to do with the crashes? Apple says USB stack is completely rewritten in El Capitan:
https://developer.apple.com/library/prerelease/mac/releasenotes/Darwin/RN_USB/Articles/10_11Changes.html

**Comment by uglyDwarf** [comment]
Lets hope the USB stack is not the culprit, otherwise my fix won't work ;)

Anyway, the backtrace pointed to the usage of FIFOs (aka named pipes), so I'm in the process of transitioning to the sockets... So far it seems it will simplify the code a bit in the end, so if it works, it will be win-win situation...

Kind regards,

Michal

**Comment by uglyDwarf** [comment]
Hello guys,
finally I managed to put the thing together ( linuxtrack.eu/test/linuxtrack-0.99.12socket.dmg ) ... So far, it seems to be working; the previous version brought the machine down in a few seconds to several minutes, but this version ran for hour and a half without a glitch...

After you install it, just make sure to run the ltr_gui first and don't forget to save the prefs...

Let me know how does it work for you.
Kind regards,

Michal

**Comment by jasso12** [comment]
So far no crashes after few hours of flying, so looking good:)

One thing I noticed after updating the X-Plane plugin is that now ltr_gui has to be active in the background for X-Plane to recognize tracking. Previously it worked without first starting ltr_gui, by simply running ltr_run command in X-Plane.

**Comment by uglyDwarf** [comment]
Can you try reinstalling the X-Plane plugin and try then? The same goes for the Linuxtrack-wine...
I'll see if I can reproduce the problem when children are in bed...

Kind regards,

Michal

**Comment by jasso12** [comment]
That was happening after I reinstalled the X-Plane plugin. First I tried with the old plugin and with that it did recognize tracking (without running ltr_gui in background), but recenter command didn't work in X-Plane.

**Comment by uglyDwarf** [comment]
I'm able to reproduce the problem; funny thing is, that it happens only on Mac, Linux runs just fine...
Will investigate deeper and provide you with a fixed version as soon as I can...

Kind regards,

Michal

**Comment by tictactoes** [comment]
Just a quick note to say that so far it's working fine in IL2 1946 with my setup with Wiiserver and Wineskin. I have not had a chance to try any longer sessions though.

**Comment by uglyDwarf** [comment]
Thank you for the report...
There is a new version available -  http://linuxtrack.eu/test/linuxtrack-0.99.12socket1.dmg
Finally this one seems to be working even in El Captain (dreaded off by one errors) - the pausing and recentering should be operational again...
Please let me know if/how does it work for you...

Kind regards,

Michal

**Comment by tictactoes** [comment]
I tried the latest version with Wiiserver/Wineskin and so far it is working fine in IL2 1946 and Rise of Flight, including pausing and re-centering. When activating tracking after a pause there seems to be a slight delay of a couple of seconds before tracking starts. I don't remember it being like that before, but I am not sure.

**Comment by jasso12** [comment]
I did some flying and I'm happy to report that now it works exactly like before I updated to El Capitan, which is to say perfectly :)
The work you are doing is invaluable, thank you again.

**Comment by tictactoes** [comment]
Yes, thank you! Where is that Donate button?

**Comment by uglyDwarf** [comment]
Thank you for those kind words... I'm glad that we were able to sort it out and thank you for your help and patience...

Kind regards,

Michal

PS. The donate button is still here: https://code.google.com/p/linux-track/
But you guys don't have to do that - you already donated your time to help improving the project...

**Comment by MinorMajor** [comment]
Thank you so much for resolving this problem, uglyDwarf.
I had been plagued by random kernel panics (Mac OS X 10.11.1 & X-Plane 10.42) and stumbled onto this thread this afternoon.  I haven't had a crash in ten flights!
I hope you posted your update to x-plane.org to help other victims of "El Capitan".
$30 donation coming your way!

**Comment by borisrorsvort** [comment]
:+1:  :clap: :clap:

**Comment by uglyDwarf** [comment]
Hello,
as a part of issue backlog cleanup, I'm going to close this issue now...

It should be fixed as of 0.99.13, which unfortunately had its own problems, but 0.99.15 should be fully operational now.

Should you encounter any more issues, please let me know...

Kind regards,

Michal
